rmf_rviz_plugin/src/RmfPanel.hpp

```diff
@@ -13,105 +13,191 @@
-
-  void set_loop_task_id(const QString& value);
-  void set_loop_start(const QString& value);
-  void set_loop_finish(const QString& value);
-  void set_loop_robot(const QString& value);
-  void set_loop_num(const QString& value);
+  void queue_delivery();
+  void queue_loop();
+  void pop_delivery();
+  void pop_plan();
+  void pop_loop();
+  void delete_plan_item();
+  void pause_robot();
+  void resume_robot();
+  void load_plan_from_file(const QString& file);
+  void open_load_file_dialog();
+  void publish_emergency_signal();
 
 protected Q_SLOTS:
-  void update_delivery_task_id();
-  void update_delivery_pickup();
-  void update_delivery_dropoff();
-  void update_delivery_robot();
-  void request_delivery();
-
-  void update_loop_task_id();
-  void update_loop_start();
-  void update_loop_finish();
-  void update_loop_robot();
-  void update_loop_num();
-  void request_loop();
+  void update_fleet_selector();
+  void update_robot_selector();
+  void update_start_waypoint_selector();
+  void update_end_waypoint_selector();
+  void update_time_selector();
+  void update_task_summary_list();
+  void update_plan();
 
 protected:
-  
   void create_layout();
+  void initialize_publishers(rclcpp::Node::SharedPtr _node);
+  void initialize_subscribers(rclcpp::Node::SharedPtr _node);
+  void initialize_state_record();
+  void initialize_qt_connections();
+  void initialize_models();
+
+  // Options - For configuring certain behaviors in the GUI
+  QCheckBox *_update_time_checkbox; // If checked, update time in _time_selector
+  QCheckBox *_pause_plan_checkbox;  // If checked, pause plan running
+  QCheckBox *_workcells_only_checkbox; // If checked, only, workcell waypoints
+                                       // will be available for selection
+  QCheckBox* _emergency_state_checkbox; // If checked, all robots to the choppa
+
+  // Selectors - For targeting agents to accomplish goals
+  QComboBox *_fleet_selector;
+  QComboBox *_robot_selector;
+  QComboBox *_start_waypoint_selector;
+  QComboBox *_end_waypoint_selector;
+  QSpinBox
+      *_repeat_count_selector; // Determines number of times to repeat an action
+  QTimeEdit *_time_selector;
+
+  // Status - For visualizing important inforation on the selected agent
+  QListView *_fleet_summary_view; // Displays task summaries from rmf_core
+  QStringListModel *_fleet_summary_model;
+  QStringList _fleet_summary_data;
+
+  // Plans - For visualizing and planning future actions
+  QListView *_plan_list_view; // Displays [action] by [fleet] at [time]
+  QPushButton *_load_action_plan_button;
+  QPushButton *_delete_plan_item_button;
+
+  QStringListModel *_plan_list_model;
+  QStringList _plan_list_data;
+  std::vector<std::pair<QTime, Delivery>> _queued_deliveries;
+  std::vector<std::pair<QTime, Loop>> _queued_loops;
+
+  // Actions - For queuing commands in Plan
+  QPushButton *_send_delivery_button;
+  QPushButton *_send_loop_button;
+  QPushButton *_pause_robot_button;
```

Comment:
ok, happy to remove this. however, would it be good to add this pause behavior eventually? it might be helpful to have this, so that users can compare their system's compliance to rmf_core's interface.

Reply:
Yes we can revisit this in the near future